pipeline/scripts/ci/upload_compose.py

```diff
@@ -0,0 +1,151 @@
+"""
+This script executes the development build upload workflow to IBM COS.
+
+The steps followed are
+- Create the necessary repo files
+- Perform a repo sync on all required repositories
+- Compress the repositories and upload them to COS
+
+Note:
+    The script is required to be executed with sudo privileges. This is required as
+    we would be creating repo files for sync operations.
+"""
+import logging
+import random
+import string
+import subprocess
+import sys
+import tempfile
+from os.path import join
+
+from docopt import docopt
+from jinja2 import Template
+
+from storage.ibm_cos import CloudObjectStorage
+
+LOG = logging.getLogger(__name__)
+REPO_TEMPLATE = """
+{%- for repo in ["OSD", "MON", "Tools"] -%}
+[{{ data.nvr }}-{{ repo }}]
+name = {{ data.nvr }}-{{ repo}}
+baseurl = {{ data.base_url }}/compose/{{ repo }}/$basearch/os
+enabled = 0
+gpgcheck = 0
+
+{% endfor %}
+"""
+
+usage = """Upload Compose.
+
+This script pulls the development RPMs available in the given base_url and uploads an
+archive of it using the provided object_name to the mentioned bucket.
+
+In the context of CephCI,
+    bucket_name     is the name of the RHCS build
+    object_name     is the ceph version
+    base_url        the base url containing the RPMs to be uploaded
+
+  Example:
+    python upload_compose.py rhceph-4.2-rhel-7 ceph-14.6.2-113 http://download.rh.com/x
```

Comment:
Is `base_url` supposed to be the base of a compose, or a MON/OSD/Tools subdirectory? Let's clarify this in the documented example here. Maybe just use a real URL here.